### PR TITLE
Add documentation for interrupt pins.

### PR DESCRIPTION
**Pull request by carlbaron**
This addresses #45.

Added some documentation for interrupt pins. Not sure if it's too detailed.

Also added an `/images` directory and a fritzing diagram. Let me know if you'd prefer me to host the image elsewhere.

Hope some of this is helpful!

[low-high-interrupts]: https://github.com/tessel/t2-firmware/blob/2230c769b6653b55c1f44581004b8c1fec531208/node/tessel-export.js#L594
@HipsterBrown

### CODE REVIEW

**Comment by HipsterBrown** [comment]
Thanks for contributing @carlbaron! I'll take a look this weekend.

File: hardware-api.md [hunk 1]
```diff
@@ -95,6 +96,83 @@ pin.analogRead(function(error, value) {
...
+has been pressed). We don't have to continuously poll to see if the voltage
+has fallen. We can just register our listener and move on.
+
+#### Available interrupt events
+
+event  | description
+-------|------------
+rise   | pin voltage rises
+fall   | pin voltage falls
+change | pin voltage changes
```
Comment (HipsterBrown) [inline hardware-api.md]: Change also passes a `value` parameter to the callback function, i.e. `pin2.on('change', function (value) {})`. So if there is a sensor attached to the interrupt, the value of the sensor will be read on `change` and passed to the callback.

**Comment by HipsterBrown** [comment]
This looks great @carlbaron! I really enjoyed the example showing how interrupts are a more efficient alternative to polling and the link to that Sparkfun article. 

One suggestion is to add a note about which pins are set as interrupts somewhere in that section. It helps to have a quick view of that information rather than continually referring to the table. 

Also pinging @Frijol for review.

**Comment by carlbaron** [comment]
Thanks @HipsterBrown! Added a new commit based on your comments.

**Comment by HipsterBrown** [comment]
Awesome! Great work again @carlbaron! LGTM :smile:

**Comment by HipsterBrown** [comment]
@carlbaron Here is an updated PNG and Fritzing to add to the PR. 

tessel-pin-interrupt

[Fritzing link](https://dl.dropboxusercontent.com/u/74986127/tessel-pin-interrupt.fzz)

**Comment by HipsterBrown** [comment]
@carlbaron Using the above PNG, please change the example event listener to use `rise`. Then download and include the Fritzing file below the example PNG. Let me know if you have any questions.

**Comment by carlbaron** [comment]
@HipsterBrown @reconbot 

Thanks for the feedback! I updated the example to use positive logic.

Thanks for supplying the new Fritzing @HipsterBrown!

**Comment by HipsterBrown** [comment]
:clap: Thanks again for your contribution @carlbaron! This is a great addition to the documentation.

**Comment by carlbaron** [comment]
Thanks @HipsterBrown! Looking forward to making further contributions! I'm going to start looking through issues, but let me know if there's anything you think I should focus on.

**Comment by HipsterBrown** [comment]
@carlbaron If you'd like to dig into the firmware code a bit, we have this open issue available -> https://github.com/tessel/t2-firmware/issues/107 

There are some comments in the issue explaining how to approach the implementation and a potential API, but feel free to reach out in the Tessel Slack if you want to pair on it or have something you'd like me to review. Also, open up the PR at any time with a `work in progress` tag. :smile:

**Comment by carlbaron** [comment]
@HipsterBrown nice I'm going to started on that button tonight!

**Comment by Frijol** [comment]
I'm late to the party on this, but: this is great! Thanks to everyone involved!

**Pull request merged**